stat/combin/combin.go

```diff
@@ -179,3 +184,115 @@ func nextCombination(s []int, n, k int) {
 		break
 	}
 }
+
+// Cartesian returns the cartesian product of the slices in data. The Cartesian
+// product of two sets is the set of all combinations of the items. For example,
+// given the input
+//  [][]float64{{1,2},{3,4},{5,6}}
+// the returned matrix will be
+//  [ 1 3 5 ]
+//  [ 1 3 6 ]
+//  [ 1 4 5 ]
+//  [ 1 4 6 ]
+//  [ 2 3 5 ]
+//  [ 2 3 6 ]
+//  [ 2 4 5 ]
+//  [ 2 4 6 ]
+// If dst is nil, a new matrix will be allocated and returned, otherwise the number
+// of rows of dst must equal \prod_i len(data[i]), and the number of columns in
+// dst must equal len(data). Cartesian also panics if len(data) = 0.
+func Cartesian(dst *mat.Dense, data [][]float64) *mat.Dense {
```

Comment:
I think an iterator API would be better here; it's not alway the case that you would want a matrix for these values, you may want to iterate over the products. So `Cartesian(fn func([]float64), data [][]float64)` might be better.

Reply:
In that package, we also have Combination and CombinationGenerator. I can add CartesianGenerator in a future PR.

Reply:
SGTM